static/src/javascripts/projects/common/modules/onward/history.spec.js

```diff
@@ -0,0 +1,395 @@
+// @flow
+import {
+    logHistory,
+    logSummary,
+    showInMegaNavEnable,
+    getPopular,
+    getContributors,
+    isRevisit,
+    reset,
+    seriesSummary,
+    mostViewedSeries,
+    _,
+} from 'common/modules/onward/history';
+
+import { local as localStorageStub } from 'lib/storage';
+
+jest.mock('lib/storage', () => ({
+    local: {
+        get: jest.fn(),
+        set: jest.fn(),
+        isAvailable: jest.fn(),
+        remove: jest.fn(),
+    },
+}));
+
+jest.mock('lib/url', () => ({
+    getPath: jest.fn(),
+}));
+
+jest.mock('fastdom');
+
+const contains = [['/p/3kvgc', 1], ['/p/3kx8f', 1], ['/p/3kx7e', 1]];
+
+const max = [
+    ['/p/3kxct', 1],
+    ['/p/3kxch', 1],
+    ['/p/3kx7n', 1],
+    ['/p/3kx8c', 1],
+    ['/p/3kx9p', 1],
+    ['/p/3kx8d', 1],
+    ['/p/3kx76', 1],
+    ['/p/3kvn8', 1],
+    ['/p/3kx7v', 1],
+    ['/p/3kx6b', 1],
+    ['/p/3kx6e', 1],
+    ['/p/3kx6n', 1],
+    ['/p/3kxvm', 1],
+    ['/p/3jqcq', 1],
+    ['/p/3kx66', 1],
+    ['/p/3kvv3', 1],
+    ['/p/3kx3k', 1],
+    ['/p/3kx2q', 1],
+    ['/p/3kvty', 1],
+    ['/p/3kxvc', 1],
+    ['/p/3kvz5', 1],
+    ['/p/3kvkt', 1],
+    ['/p/3kvjk', 1],
+    ['/p/3kvgc', 1],
+    ['/p/3kvc6', 1],
+    ['/p/3kvdv', 1],
+    ['/p/3kvbh', 1],
+    ['/p/3kvct', 1],
+    ['/p/3kv7v', 1],
+    ['/p/3kvxk', 1],
+    ['/p/3kv9f', 1],
+    ['/p/3kv9g', 1],
+    ['/p/3jqcp', 1],
+    ['/p/3kv5t', 1],
+    ['/p/3kv7b', 1],
+    ['/p/3kv5j', 1],
+    ['/p/3kv6h', 1],
+    ['/p/3kvx8', 1],
+    ['/p/3jpnc', 1],
+    ['/p/3jqnc', 1],
+    ['/p/3jqm3', 1],
+    ['/p/3jqzk', 1],
+    ['/p/3jqz5', 1],
+    ['/p/3jqyf', 1],
+    ['/p/3jqzg', 1],
+    ['/p/3jqmn', 1],
+    ['/p/3jqk5', 1],
+    ['/p/3jqha', 1],
+    ['/p/3jqg5', 1],
+    ['/p/3jqf9', 1],
+    ['/p/g7sf9', 1],
+];
```

Comment:
Sure, I hadn't seen this lib in use before but if it'll reduce the test boilerplate then that's :ok_hand: with me.